pkg/controllers/user/nodesyncer/cordonfieldssyncer.go

```diff
@@ -26,10 +36,130 @@ func (m *NodesSyncer) syncCordonFields(key string, obj *v3.Node) error {
 			return err
 		}
 	}
+	if !desiredValue {
+		removeDrainCondition(obj)
```

Comment:
looks like we are modifying the object directly in this method; it can affect cache. See if you need to make deep copy before